corehq/apps/ota/views.py

```diff
@@ -139,21 +139,31 @@ def claim(request, domain):
     as_user_obj = CouchUser.get_by_username(as_user) if as_user else None
     restore_user = get_restore_user(domain, request.couch_user, as_user_obj)
 
-    case_id = unquote(request.POST.get('case_id', ''))
-    if not case_id:
+    # This will be expecting space seperated values of all the cases to be claimed
+    case_ids = request.POST.getlist('case_id')
+
+    if not case_ids:
         return HttpResponse('A case_id is required', status=400)
 
     try:
-        if get_first_claim(domain, restore_user.user_id, case_id):
-            return HttpResponse(status=204)
+        case_ids_already_claimed = get_first_claims(domain, restore_user.user_id, case_ids)
+        if case_ids_already_claimed:
+            case_ids_to_claim = [case for case in case_ids if case not in case_ids_already_claimed]
```

Comment:
A lot of this code is conceptually set logic (you use `set()` down below too) so you could simplify it a little by coercing to a set up-front (including `case_ids`)
```suggestion
        case_ids_already_claimed = set(get_first_claims(domain, restore_user.user_id, case_ids))
        if case_ids_already_claimed:
            case_ids_to_claim = case_ids - case_ids_already_claimed
```
In fact, you might also change `get_first_claims` to return a `set`